src/ray/protobuf/raylet.proto

```diff
@@ -0,0 +1,205 @@
+syntax = "proto3";
+
+package ray.rpc;
+
+import "src/ray/protobuf/common.proto";
+import "src/ray/protobuf/gcs.proto";
+
+// Service request and reply message
+message RegisterClientRequest {
+  // Indicates the client is a worker or a driver.
+  bool is_worker = 1;
+  // The worker id.
+  bytes worker_id = 2;
+  // The process ID of this worker.
+  uint32 worker_pid = 3;
+  // The job ID.
+  bytes job_id = 4;
+  // Language of this worker.
+  Language language = 5;
+  // Port that this worker is listening on.
+  // If port > 0, then worker will listen to this port and wait for
+  // raylet to push tasks, instead of invoking GetTask().
+  int32 port = 6;
+}
+message RegisterClientReply {
+  repeated int32 gpu_ids = 1;
+}
+
+message SubmitTaskRequest {
+  bytes task_spec = 1;
+  repeated bytes execution_dependencies = 2;
+}
+message SubmitTaskReply {
+}
+
+message DisconnectClientRequest {
+  bytes worker_id = 1;
+}
+message DisconnectClientReply {
+}
+
+message GetTaskRequest {
+  bytes worker_id = 1;
+}
+message GetTaskReply {
+  // A string of bytes representing the task specification.
+  bytes task_spec = 1;
+  // A list of the resources reserved for this worker.
+  repeated ResourceIdSetInfo fractional_resource_ids = 2;
+}
+
+message TaskDoneRequest {
+  bytes worker_id = 1;
+}
+message TaskDoneReply {
+}
+
+message FetchOrReconstructRequest {
+  // List of object IDs of the objects that we want to reconstruct or fetch.
+  repeated bytes object_ids = 1;
+  // Do we only want to fetch the objects or also reconstruct them?
+  bool fetch_only = 2;
+  // The current task ID. If fetch_only is false, then this task is blocked.
+  bytes task_id = 3;
+  // The worker ID.
+  bytes worker_id = 4;
+}
+message FetchOrReconstructReply {
+}
+
+message NotifyUnblockedRequest {
+  bytes worker_id = 1;
+  // The current task ID. This task is no longer blocked.
+  bytes task_id = 2;
+}
+message NotifyUnblockedReply {
+}
+
+message WaitRequest {
+  // List of object ids we'll be waiting on.
+  repeated bytes object_ids = 1;
+  // Number of objects expected to be returned, if available.
+  uint64 num_ready_objects = 2;
+  // timeout
+  int64 timeout = 3;
+  // Whether to wait until objects appear locally.
+  bool wait_local = 4;
```

Comment:
wait_local means wait for a local objects, all variables related to this in raylet is wait_local. And the variable in prevous raylet client is also wait_local, I think we should keep it unchanged.

> "wait_local" is somewhat unclear to me - sounds like it might specify if the client or server does the waiting or something like that.

wait_local means wait for a local objects, all variables related to this in raylet is wait_local. And the variable in prevous raylet client is also wait_local, I think we should keep it unchanged.